zerver/webhooks/semaphore/view.py

```diff
@@ -22,59 +25,186 @@
+({repository_url}):
+* **Tag**: [{tag_name}]({tag_url})
+* **Pipeline**:
+  - Name: {pipeline_name}
+  - Result: {pipeline_result}
+* **Blocks**:
+{blocks}
+""".strip()
+
+DEFAULT_TEMPLATE = """
+An activity occured on branch **{branch_name}** of \
+[{organization_name}/{project_name}]({repository_url}):
+* **Author**: []{username}]({profile_url})
+* **Pipeline**:
+  - Name: {pipeline_name}
+  - Result: {pipeline_result}
+* **Blocks**:
+{blocks}
+""".strip()
+
 TOPIC_TEMPLATE = "{project}/{branch}"
 
 @api_key_only_webhook_view('Semaphore')
 @has_request_variables
 def api_semaphore_webhook(request: HttpRequest, user_profile: UserProfile,
                           payload: Dict[str, Any]=REQ(argument_type='body')) -> HttpResponse:
 
-    # semaphore only gives the last commit, even if there were multiple commits
-    # since the last build
-    branch_name = payload["branch_name"]
-    project_name = payload["project_name"]
-    result = payload["result"]
-    event = payload["event"]
-    commit_id = payload["commit"]["id"]
-    commit_url = payload["commit"]["url"]
-    author_email = payload["commit"]["author_email"]
-    message = payload["commit"]["message"]
-
-    if event == "build":
-        build_url = payload["build_url"]
-        build_number = payload["build_number"]
-        content = BUILD_TEMPLATE.format(
-            build_number=build_number,
-            build_url=build_url,
-            status=result,
-            commit_hash=commit_id[:7],
-            commit_message=message,
-            commit_url=commit_url,
-            email=author_email
-        )
-
-    elif event == "deploy":
-        build_url = payload["build_html_url"]
-        build_number = payload["build_number"]
-        deploy_url = payload["html_url"]
-        deploy_number = payload["number"]
-        server_name = payload["server_name"]
-        content = DEPLOY_TEMPLATE.format(
-            deploy_number=deploy_number,
-            deploy_url=deploy_url,
-            build_number=build_number,
-            build_url=build_url,
-            status=result,
-            commit_hash=commit_id[:7],
-            commit_message=message,
-            commit_url=commit_url,
-            email=author_email,
-            server_name=server_name
-        )
-
-    else:  # should never get here
-        content = "{event}: {result}".format(
-            event=event, result=result)
+    if 'event' in payload.keys():
+        # semaphore only gives the last commit, even if there were multiple commits
+        # since the last build
+        branch_name = payload["branch_name"]
+        project_name = payload["project_name"]
+        result = payload["result"]
+        event = payload["event"]
+        commit_id = payload["commit"]["id"]
+        commit_url = payload["commit"]["url"]
+        author_email = payload["commit"]["author_email"]
+        message = payload["commit"]["message"]
```

Comment:
There are many missing fields which are producing `missing_key` errors, should I assign default values in case of missing field or should I change the notification content so that these variables are not used and shift the statements fetching these variables from payload inside the existing `if` conditions which are using these variables.